### ISSUE TITLE
Fix S1168 FN: Support mathematical operators in generic types

### ISSUE THREAD

**Issue by cristian-ambrosini-sonarsource** [SonarSource/sonar-dotnet#6393]
### Description

FN for S1168 (ReturnEmptyCollectionInsteadOfNull).

Using C#11 static mathematical operator overload, the rule is not raising an issue.

### Repro steps

```csharp
class IInterface : IAdditionOperators<IInterface, IInterface, IEnumerable<string>>
{
    public static IEnumerable<string> operator +(IInterface left, IInterface right) => null; // FN
}
```

### Expected behavior

The rule should raise an issue.

### Actual behavior

The rule does not raise an issue.
